### PR TITLE
Issue #sb-forms : allow ecm fix

### PR DESCRIPTION
**Pull request by pradeepkumarcm96**
###

### CODE REVIEW

File: projects/common-form-elements/src/lib/keywords/keywords.component.ts [hunk 1]
```diff
@@ -1,35 +1,100 @@
...
+      (response: any) => {
+        if (response && response.options) {
+          this.field.options = response.options;
+        } else {
+          this.field.options = null;
+        }
+      }
+    );
+  }
+console.log(this.formControlRef.isVisible);
```
Comment (venkateshwarans) [inline projects/common-form-elements/src/lib/keywords/keywords.component.ts:79]: remove the console.log

File: projects/common-form-elements/src/lib/keywords/keywords.component.ts [hunk 2]
```diff
@@ -1,35 +1,100 @@
...
+      this.options = _.isEmpty(this.field.options) ? this.isOptionsClosure(this.field.options) && this.field.options : [];
+    }
+
+    if (!_.isEmpty(this.depends) && !this.isOptionsClosure(this.options)) {
+      this.contextValueChangesSubscription =  merge(..._.map(this.depends, depend => depend.valueChanges)).pipe(
+       tap((value: any) => {
+         this.latestParentValue = value;
+       })
+       ).subscribe();
+     }
```
Comment (venkateshwarans) [inline projects/common-form-elements/src/lib/keywords/keywords.component.ts:56]: are we making use of `this.latestParentValue` value anywhere? if not please remove this entire if condition for now and add the `this.contextValueChangesSubscription` to line.no 59.

**Review by venkateshwarans — CHANGES_REQUESTED** [review]
see comments

File: projects/common-form-elements/src/lib/keywords/keywords.component.ts [hunk 3]
```diff
@@ -1,35 +1,92 @@
...
+          tap(() => {
+            this.formControlRef.patchValue(null);
+          })
+      ).subscribe();
+  }
+
+  if (this.isOptionsClosure(this.options)) {
+    // tslint:disable-next-line:max-line-length
+    this.options$ = (this.options as DynamicFieldConfigOptionsBuilder<any>)(this.formControlRef, this.depends, this.formGroup, () => this.dataLoadStatusDelegate.next('LOADING'), () => this.dataLoadStatusDelegate.next('LOADED')) as any;
+    this.options$.subscribe(
```
Comment (venkateshwarans) [inline projects/common-form-elements/src/lib/keywords/keywords.component.ts:62]: add `contextValueChangesSubscription` here

**Review by venkateshwarans — APPROVED** [review]

**Review by pallakartheekreddy — APPROVED** [review]

**Pull request merged**